include/onnxruntime/core/session/onnxruntime_c_api.h

```diff
@@ -1146,6 +1146,9 @@ struct OrtApi {
   */
   ORT_API2_STATUS(ModelMetadataGetGraphDescription, _In_ const OrtModelMetadata* model_metadata,
                   _Inout_ OrtAllocator* allocator, _Outptr_ char** value);
+
+  ORT_API2_STATUS(SetCurrentGpuDeviceId, _In_ int device_id);
```

Comment:
Shouldn't this mention cuda somewhere to differentiate it from ROCM devices? Or is the plan to make it switch which underlying method is called if ROCM is enabled?

Reply:
Let me make it explicit here.

Reply:
In my opinion, it is OK to be generic here since ROCM and CUDA won't co-exist. Otherwise, another almost duplicate APIs need to be added for ROCM.